.circleci/config.yml

```diff
@@ -1,22 +1,161 @@
 version: 2.1
 
-jobs:
-  test:
+commands:
+  install_dependencies:
+    description: "Install development dependencies."
+    steps:
+      - run: composer install
+  mkdir_artifacts:
+    description: "Make Artifacts directory"
+    steps:
+      - run:
+          command: |
+            [ ! -d "/tmp/artifacts" ] && mkdir /tmp/artifacts &>/dev/null
+  svn_setup:
+    description: "Setup SVN"
+    steps:
+      - mkdir_artifacts
+      - run:
+          command: |
+            cd /tmp/artifacts
+            SLUG=grep 'Text Domain:' /tmp/src/plugin.php | awk -F: '{print $2}' | sed 's/^\s//'
+            svn co https://plugins.svn.wordpress.org/$SLUG --depth=empty .
+            svn up trunk
+            svn up tags --depth=empty
+            cd trunk
+            rm -rf * .* *.*
+            cp -r /tmp/src/* .
+            svn propset svn:ignore -F .svnignore .
+  svn_add_changes:
+    description: "Add changes to SVN"
+    steps:
+      - deploy:
+          command: |
+            cd /tmp/artifacts
+            svn st | grep ^! | awk '{print " --force "$2}' | xargs svn rm
+            svn add --force .
+  svn_create_tag:
+    description: "Create a SVN tag"
+    steps:
+      - deploy:
+          command: |
+            cd /tmp/artifacts
+            VERSION=grep 'Version:' /tmp/src/plugin.php | awk -F: '{print $2}' | sed 's/^\s//'
+            svn cp trunk tags/$VERSION
+  svn_commit:
+    description: "Commit changes to SVN"
+    steps:
+      - deploy:
+          command: |
+            cd /tmp/artifacts
+            VERSION=grep 'Version:' /tmp/src/plugin.php | awk -F: '{print $2}' | sed 's/^\s//'
+            svn ci -m "Tagging $VERSION from Github" --no-auth-cache --non-interactive --username "$SVN_USERNAME" --password "$SVN_PASSWORD"
+
+executors:
+  base:
+    docker:
+      - image: circleci/buildpack-deps:latest
+    working_directory: /tmp
+  php_node:
     docker:
       - image: circleci/php:7.3.3-stretch-node-browsers
+    working_directory: /tmp/src
+
+jobs:
+  checkout:
+    executor: base
     steps:
-      - checkout
-      - prepare-environment
+      - checkout:
+          path: src
+      - persist_to_workspace:
+          root: /tmp
+          paths:
+            - src
+
+  checks:
+    executor: php_node
+    steps:
+      - attach_workspace:
+          at: /tmp
+      - install_dependencies
```

Comment:
Is there a benefit to breaking out small runs like 'composer install' to its own command, instead of doing the 'run: ' here?

Reply:
For just `composer install`, not really. However I'm not sure if other plugins have additional package managers or build steps. Splitting it out makes it easier to add as needed.